packages/web3-providers-ws/src/index.ts

```diff
@@ -0,0 +1,387 @@
+						requestItem?.deferredPromise.resolve(response);
+					} else if ('error' in response && response.error !== undefined) {
+						this.wsEventEmitter.emit('message', response, null);
+						requestItem?.deferredPromise.reject(response);
+					}
+
+					this.sentQueue.delete(id);
+				}
+			},
+		);
+	}
+
+	private onConnect() {
+		this.providerStatus = 'connected';
+		this.reconnectAttempts = 0;
+
+		if (this.requestQueue.size > 0) {
+			for (const value of this.requestQueue.values()) {
+				// eslint-disable-next-line
+				this.request(value.payload);
+			}
+		}
+	}
+
+	private onClose(event: ConnectionEvent): void {
+		this.providerStatus = 'disconnected';
+		if (
+			this.wsProviderOptions?.reconnectOptions?.auto &&
+			(![1000, 1001].includes(event.code) || event.wasClean === false)
+		) {
+			this.reconnect();
+			return;
+		}
+
+		if (this.requestQueue.size > 0) {
+			this.requestQueue.forEach((request: WSRequestItem, key: JsonRpcId) => {
+				request.deferredPromise.reject(new ConnectionNotOpenError(event));
+				this.requestQueue.delete(key);
+			});
+		}
+
+		if (this.sentQueue.size > 0) {
+			this.sentQueue.forEach((request: WSRequestItem, key: JsonRpcId) => {
+				request.deferredPromise.reject(new ConnectionNotOpenError(event));
+				this.sentQueue.delete(key);
+			});
+		}
+
+		this.removeSocketListeners();
+	}
+
+	private parseResponse(dataReceived: string): JsonRpcResponse[] {
+		const returnValues: JsonRpcResponse[] = [];
+
+		// de chunk
+		const dechunkedData = dataReceived
+			.replace(/\}[\n\r]?\{/g, '}|--|{') // }{
+			.replace(/\}\][\n\r]?\[\{/g, '}]|--|[{') // }][{
+			.replace(/\}[\n\r]?\[\{/g, '}|--|[{') // }[{
+			.replace(/\}\][\n\r]?\{/g, '}]|--|{') // }]{
+			.split('|--|');
+
+		dechunkedData.forEach((data: string) => {
+			let dataToParse: string = data;
+
+			if (this.lastDataChunk !== '') dataToParse = this.lastDataChunk + dataToParse;
+
+			let result: JsonRpcResponse;
+
+			try {
+				// eslint-disable-next-line
+				result = JSON.parse(dataToParse);
+			} catch (error) {
+				this.lastDataChunk = dataToParse;
+
+				if (this.lastChunkTimeout) clearTimeout(this.lastChunkTimeout);
+
+				this.lastChunkTimeout = setTimeout(() => {
+					if (
+						this.wsProviderOptions.reconnectOptions?.auto &&
+						this.wsProviderOptions.reconnectOptions?.onTimeout
+					) {
+						this.reconnect();
+						return;
+					}
+
+					if (this.requestQueue.size > 0) {
+						this.requestQueue.forEach((request: WSRequestItem, key: JsonRpcId) => {
+							request.deferredPromise.reject(
+								new ConnectionTimeoutError(this.wsProviderOptions.customTimeout),
+							);
+							this.requestQueue.delete(key);
+						});
+					}
+				}, this.wsProviderOptions.customTimeout);
+
+				return;
+			}
+
+			clearTimeout(this.lastChunkTimeout);
+			this.lastDataChunk = '';
+
+			if (result) returnValues.push(result);
+		});
+
+		return returnValues;
+	}
+
+	private removeSocketListeners(): void {
+		// eslint-disable-next-line
```

Comment:
I think an explanation (like error message + reasons why it's not being fixed) should be provided when disabling linting. Additionally, you should be able to disable the specific rule for the line instead of disabling linting entirely